flink-connector/src/main/java/com/datastrato/gravitino/flink/connector/hive/GravitinoHiveCatalogFactory.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.flink.connector.hive;
+
+import com.datastrato.gravitino.flink.connector.utils.PropertyUtils;
+import java.util.Set;
+import org.apache.flink.configuration.ConfigOption;
+import org.apache.flink.table.catalog.Catalog;
+import org.apache.flink.table.catalog.hive.HiveCatalog;
+import org.apache.flink.table.catalog.hive.factories.HiveCatalogFactory;
+import org.apache.flink.table.catalog.hive.factories.HiveCatalogFactoryOptions;
+import org.apache.flink.table.factories.CatalogFactory;
+import org.apache.flink.table.factories.FactoryUtil;
+import org.apache.hadoop.hive.conf.HiveConf;
+
+/**
+ * Factory for creating instances of {@link GravitinoHiveCatalog}. It will be created by SPI
+ * discovery in Flink.
+ */
+public class GravitinoHiveCatalogFactory implements CatalogFactory {
+  /** Identifier for the {@link GravitinoHiveCatalog}. */
+  public static final String IDENTIFIER = "gravitino-hive";
+
+  private HiveCatalogFactory hiveCatalogFactory;
+
+  @Override
+  public Catalog createCatalog(Context context) {
+    this.hiveCatalogFactory = new HiveCatalogFactory();
+    final FactoryUtil.CatalogFactoryHelper helper =
+        FactoryUtil.createCatalogFactoryHelper(this, context);
+    helper.validateExcept(
+        PropertyUtils.HIVE_PREFIX,
+        PropertyUtils.HADOOP_PREFIX,
+        PropertyUtils.DFS_PREFIX,
+        PropertyUtils.FS_PREFIX);
+
+    String hiveConfDir = helper.getOptions().get(HiveCatalogFactoryOptions.HIVE_CONF_DIR);
```

Comment:
hive-conf-dir is required for now. We can also choose to load all properties from the gravitino and skip the validation. What do you think?

Reply:
Another way we can remove the validation and keep the hive-conf-dir and hadoop-conf-dir optional.

Reply:
if `hiveConfDir` or `hadoopConfDir` is null, this will cause exception, right?

Reply:
ok